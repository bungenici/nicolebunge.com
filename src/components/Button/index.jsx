import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

class Button extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    to: PropTypes.string.isRequired,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children, to } = this.props;

    return (
      <Link
        className={styles.button}
        to={to}
      >
        {children}
      </Link>
    );
  }
}

export default Button;
