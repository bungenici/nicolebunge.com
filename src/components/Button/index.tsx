import classNames from 'classnames';
import { Link } from 'gatsby';
import React from 'react';
import styles from './style.module.css';

export interface ButtonProps {
  className?: string;
  to: string;
}

const Button: React.FC<ButtonProps> = (props) => {
  const { children, className, to, ...otherProps } = props;

  return (
    <Link className={classNames(className, styles.button)} to={to} {...otherProps}>
      {children}
    </Link>
  );
};

export default Button;
