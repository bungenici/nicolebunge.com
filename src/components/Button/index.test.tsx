/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';

describe('Button component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button to="/">Button</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
