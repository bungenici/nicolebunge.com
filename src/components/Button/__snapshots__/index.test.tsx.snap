// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<DocumentFragment>
  <a
    class="button"
    href="/"
  >
    Button
  </a>
</DocumentFragment>
`;
